### PR TITLE
Part 2: add ability to edit instructions

### PR DESCRIPTION
**Pull request by jlhogantech**
resolves https://github.com/HHS/simpler-grants-pdf-builder/issues/484

- Fix styles for edit view instructions, and add ability to edit
- add a fix for matching instructions to subsections without a name, and a test for the case

| Composer edit view with instructions | Composer instructions edit view | 
| --- | --- | 
| Screenshot 2025-10-30 194835 | Screenshot 2025-10-30 194848 |

### CODE REVIEW

File: nofos/nofos/test_nofo.py
Comment (jlhogantech) [inline nofos/nofos/test_nofo.py]: Most of these changes are just whitespace/formatting, sorry! Turns out I did not actually run the precommit stuff on my prev PR 😞 did add one net new test for the no-name subsection, which I will call out with a comment below

File: nofos/nofos/test_nofo.py [hunk 1]
```diff
@@ -6895,31 +6901,65 @@ def test_add_instructions_to_subsections_matches_subsequent_duplicates(self):
...
+        assert (
+            "Instructions for Section B"
+            in sections[0]["subsections"][1]["instructions"]
+        )
+        assert (
+            "Instructions for Section A"
+            in sections[0]["subsections"][2]["instructions"]
+        )
+
+    def test_add_instructions_to_subsections_skips_subsections_with_no_title(self):
```
Comment (jlhogantech) [inline nofos/nofos/test_nofo.py]: this is the actual change -- new test added for additional edge case

**Comment by pcraig3** [comment]
> Currently, the instructions are HTML, which is formatted nicely by the martor preview stuff but maybe harder for content guide writers to edit. Should the instructions HTML be parsed into markdown before exposing to users?

Yes, HTML is a good markup language for rendering content but it is horrible for editing. 

We have this same issue with the subsection `body`, which comes in as HTML and we convert to markdown. 

The code where we handle this is here: https://github.com/HHS/simpler-grants-pdf-builder/blob/main/nofos/nofos/nofo.py#L374-L387

So let's do the same thing for instructions

The conversion back from markdown should already be working because of the `safe_markdown` tag we are using in the template.

**Comment by pcraig3** [comment]
> Currently, the instructions are HTML, which is formatted nicely by the martor preview stuff but maybe harder for content guide writers to edit. Should the instructions HTML be parsed into markdown before exposing to users?

Actually, I didn't realize this before, but these instructions are being saved as tables because that's how they are structured in the Word document (they are a 1-cell table in the Word docs).

I think we should be stripping the table semantics away, because it is not necessary and will make it confusing to edit in markdown. A 1-cell table is a shortcut we use in Word so that it is easy for the NOFO Builder to find these instructions and remove them. It has no semantic meaning and it shouldn't be a table.

//

Secondly, I am wondering if we should also delete the first line of the Instructions and just construct the "Instructions for NOFO Writers: {subsection.name}" ourselves using the subsection name. There are definitely some edge cases here, however, as not all subsections have names, and some Instructions titles include more than 1 subsection, so we would be changing the intent if the heading changed. On this aspect of things I am not sure what the right way to play it is, I could be swayed in either direction.

**Review by pcraig3 — CHANGES_REQUESTED** [review]
Not that much from me, but I have a comment outside of this review about stripping the table semantics out from the instructions and then converting its contents to markdown.

**Comment by jlhogantech** [comment]
Most recent change:
1. Adds instructions_edit to css rules for correct font family 
2. Extracts instructions as markdown, removing the wrapping one-cell table

does NOT extract instructions title as separate, uneditable field because I'm pretty sure that would require a migration and is a change to the existing ContentGuideSubsection model, and i wanted to confirm that's what we want to do

**Review by pcraig3 — APPROVED** [review]
It's looking good to me! The only question/comment I had was around instructions titles, so I left a thorough comment about it above. ☝️ 

In the end, I am okay with whatever the call is on this, so merge when you are ready.
